### PR TITLE
digibyte: 7.17.2 -> 7.17.3

### PR DESCRIPTION
**Pull request by centromere**
###### Motivation for this change
Update to 7.17.3.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131007` at a124f4d8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>digibyte</li>
    <li>digibyted</li>
  </ul>
</details>

File: pkgs/applications/blockchains/digibyte.nix [hunk 1]
```diff
@@ -1,33 +1,21 @@
...
-, protobuf
-, hexdump
-, zeromq
-, withGui
-, qtbase ? null
-, qttools ? null
-, wrapQtAppsHook ? null
+{ lib, stdenv, fetchFromGitHub, openssl, boost, libevent, autoreconfHook
+, db4, pkg-config, protobuf, hexdump, zeromq, withGui ? true, qtbase ? null
+, qttools ? null, wrapQtAppsHook ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/digibyte.nix:3]:
```suggestion
, db4, pkg-config, protobuf, hexdump, zeromq, withGui ? true, qtbase
, qttools, wrapQtAppsHook
```

File: pkgs/applications/blockchains/digibyte.nix [hunk 2]
```diff
@@ -59,6 +47,16 @@ stdenv.mkDerivation rec {
       "--with-qt-bindir=${qtbase.dev}/bin:${qttools.dev}/bin"
   ];
 
+  postInstall = ''
+    rm $out/bin/test_* && \
+    rm $out/bin/bench_*
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/digibyte.nix:52]:
```suggestion
    rm $out/bin/test_* $out/bin/bench_*
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/blockchains/digibyte.nix (hunk 1 above)
Comment (centromere) [reply pkgs/applications/blockchains/digibyte.nix]: The following lines exist in [all-packages.nix](https://raw.githubusercontent.com/NixOS/nixpkgs/34425dac6f28df548ef378096a09e3389791de5a/pkgs/top-level/all-packages.nix):
```nix
digibyte = libsForQt514.callPackage ../applications/blockchains/digibyte.nix { withGui = true; };
digibyted = callPackage ../applications/blockchains/digibyte.nix { withGui = false; };
```

As a result, `qtbase` [can be null](https://github.com/NixOS/nixpkgs/runs/3135375532?check_suite_focus=true#step:4:113). What do you think is the best way to proceed?

Comment (SuperSandro2000) [reply pkgs/applications/blockchains/digibyte.nix]: It cannot be null. The package is still defined.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -28545,8 +28545,8 @@ in
 
   dero = callPackage ../applications/blockchains/dero.nix { boost = boost165; };
 
-  digibyte = libsForQt514.callPackage ../applications/blockchains/digibyte.nix { withGui = true; };
-  digibyted = callPackage ../applications/blockchains/digibyte.nix { withGui = false; };
+  digibyte  = libsForQt514.callPackage ../applications/blockchains/digibyte.nix { withGui = true; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:28548]: Can we update this?

File: pkgs/applications/blockchains/digibyte.nix [hunk 4]
```diff
@@ -1,33 +1,21 @@
...
 }:
 
 with lib;
 
 stdenv.mkDerivation rec {
   pname = "digibyte";
-  version = "7.17.2";
+  version = "7.17.3";
 
   name = pname + toString (optional (!withGui) "d") + "-" + version;
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/digibyte.nix:13]:
```suggestion
```

File: pkgs/applications/blockchains/digibyte.nix [hunk 5]
```diff
@@ -1,33 +1,21 @@
...
-, wrapQtAppsHook ? null
+{ lib, stdenv, fetchFromGitHub, openssl, boost, libevent, autoreconfHook
+, db4, pkg-config, protobuf, hexdump, zeromq, withGui ? true, qtbase
+, qttools, wrapQtAppsHook
 }:
 
 with lib;
 
 stdenv.mkDerivation rec {
   pname = "digibyte";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/digibyte.nix:9]:
```suggestion
  pname = digibyte${lib.optionalString (!withGui) "d"}";

```

File: pkgs/applications/blockchains/digibyte.nix [hunk 6]
```diff
@@ -1,33 +1,21 @@
...
-, withGui
-, qtbase ? null
-, qttools ? null
-, wrapQtAppsHook ? null
+{ lib, stdenv, fetchFromGitHub, openssl, boost, libevent, autoreconfHook
+, db4, pkg-config, protobuf, hexdump, zeromq, withGui ? true, qtbase
+, qttools, wrapQtAppsHook
 }:
 
 with lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/digibyte.nix:7]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**